### PR TITLE
MNT New tests classes to test version tags

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Description
Additional classes for creating objects for testing recursive checking of changes in child classes.
## Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1593

### CODE REVIEW

File: code/versioned/NonVersionedParentObject.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace SilverStripe\FrameworkTest\Versioned\Model;
+
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\ORM\DataObject;
+
+class NonVersionedParentObject extends DataObject
+{
+    private static $table_name = 'NonVersionedParentObject';
```
Comment (emteknetnz) [inline code/versioned/NonVersionedParentObject.php]:
```suggestion
    private static $table_name = 'FrameworkTest_NonVersionedParentObject';
```

File: code/versioned/NonVersionedChildObject.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace SilverStripe\FrameworkTest\Versioned\Model;
+
+use SilverStripe\ORM\DataObject;
+
+class NonVersionedChildObject extends DataObject
+{
+    private static $table_name = 'NonVersionedChildObject';
```
Comment (emteknetnz) [inline code/versioned/NonVersionedChildObject.php]:
```suggestion
    private static $table_name = 'FrameworkTest_NonVersionedChildObject';
```

File: code/versioned/VersionedChildObject.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace SilverStripe\FrameworkTest\Versioned\Model;
+
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class VersionedChildObject extends DataObject
+{
+    private static $table_name = 'VersionedChildObject';
```
Comment (emteknetnz) [inline code/versioned/VersionedChildObject.php]:
```suggestion
    private static $table_name = 'FrameworkTest_VersionedChildObject';
```

File: code/versioned/TestPageVersionedObject.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+use Page;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+
+/**
+ * Parent class of all test pages
+ */
+class TestPageVersionedObject extends Page implements \TestPageInterface
+{
+    private static $table_name = 'FrameworkTestPage_VersionedObject';
```
Comment (emteknetnz) [inline code/versioned/TestPageVersionedObject.php]:
```suggestion
    private static $table_name = 'FrameworkTest_VersionedObject';
```

File: code/versioned/VersionedParentObject.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+use Page;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class VersionedParentObject extends DataObject
+{
+    private static $table_name = 'VersionedParentObject';
```
Comment (emteknetnz) [inline code/versioned/VersionedParentObject.php]:
```suggestion
    private static $table_name = 'FrameworkTest_VersionedParentObject';
```

File: code/versioned/TestPageVersionedObject.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+namespace SilverStripe\FrameworkTest\Versioned\Model;
+
+use Page;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+
+/**
+ * Parent class of all test pages
+ */
+class TestPageVersionedObject extends Page implements \TestPageInterface
```
Comment (emteknetnz) [inline code/versioned/TestPageVersionedObject.php]:
```suggestion
class TestPageVersionedObject extends Page implements TestPageInterface
```

And add `use TestPageInterface;` above

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: code/versioned/NonVersionedParentObject.php (hunk 1 above)
Comment (sabina-talipova) [reply code/versioned/NonVersionedParentObject.php]: Updated

File: code/versioned/NonVersionedChildObject.php (hunk 2 above)
Comment (sabina-talipova) [reply code/versioned/NonVersionedChildObject.php]: Updated

File: code/versioned/VersionedParentObject.php (hunk 5 above)
Comment (sabina-talipova) [reply code/versioned/VersionedParentObject.php]: Updated

File: code/versioned/TestPageVersionedObject.php (hunk 6 above)
Comment (sabina-talipova) [reply code/versioned/TestPageVersionedObject.php]: Updated

File: code/versioned/TestPageVersionedObject.php (hunk 4 above)
Comment (sabina-talipova) [reply code/versioned/TestPageVersionedObject.php]: Updated

**Comment by GuySartorelli** [comment]
Closing - we can reference this PR if we decide to go forward with this work in the future, pending the spike.

**Pull request closed without merging**